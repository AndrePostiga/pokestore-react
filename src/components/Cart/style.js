import styled from 'styled-components';
import { darken } from 'polished';

export const Container = styled.div`
  width: 30%;
  @media (max-width: 900px) {
    width: 100%;
    margin-bottom: 10px;
    margin-left: 0;
  }

  border-radius: 4px;
  background-color: #fff;
  margin-left: 20px;
  padding: 20px;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
`;

export const ProductCartView = styled.ul`
  display: flex;
  flex-direction: column;
  list-style: none;
  height: 350px;
  overflow: auto;

  @media (max-width: 900px) {
    height: 200px;
  }

  li {
    background: #fff;
    display: flex;
    align-items: center;
    justify-content: space-between;
    padding: 10px;

    & + li {
      padding-top: 15px;
      border-top: 1px solid rgba(0, 0, 0, 0.1);
      @media (max-width: 900px) {
        padding-top: 5px;
      }
    }

    img {
      max-width: 60px;

      @media (max-width: 900px) {
        max-width: 30px;
      }
    }

    button {
      background: red;
      border: 0;
      border-radius: 4px;
      overflow: hidden;
      padding: 20px;

      @media (max-width: 900px) {
        padding: 5px;
      }

      transition: background 0.2s;

      &:hover {
        background: ${darken(0.03, 'red')};
      }
    }
  }
`;

export const CartFooter = styled.div`
  div {
    padding: 20px 0;
    display: flex;
    justify-content: space-between;
    align-items: baseline;
    border-top: 1px solid rgba(0, 0, 0, 0.1);
    border-bottom: 1px solid rgba(0, 0, 0, 0.1);

    strong {
      font-size: 20px;
      line-height: 20px;
      color: #333;
    }

    span {
      font-size: 20px;
      line-height: 20px;
      font-weight: bold;
    }
  }

  button {
    background: ${(props) => props.theme};
    color: #fff;
    border: 0;
    border-radius: 4px;
    width: 100%;
    overflow: hidden;
    padding: 12px;
    transition: background 0.2s;

    &:hover {
      background: ${(props) => darken(0.03, props.theme)};
    }

    span {
      flex: 1;
      text-align: center;
      font-weight: bold;
    }
  }
`;
