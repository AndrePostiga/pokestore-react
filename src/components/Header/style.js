import styled from 'styled-components';

export const Container = styled.div`
  background: orange;
`;

export const Navigation = styled.nav`
  background: ${(props) => props.theme};
  display: flex;
  justify-content: baseline;
  width: 100% !important;
  padding: 20px 30px;

  input {
    width: 100%;
    border: 1px solid #ddd;
    border-radius: 4px;
    color: #666;
    margin-left: 20px;
    box-sizing: border-box;
    padding-left: 20px;
  }

  button {
    background: ${(props) => props.theme};
    border: 0;
    border-radius: 4px;
    overflow: hidden;
  }
`;
