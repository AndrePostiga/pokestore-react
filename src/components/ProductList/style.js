import styled from 'styled-components';
import { darken } from 'polished';

export const Container = styled.div`
  width: 70%;
  @media (max-width: 900px) {
    width: 100%;
  }
`;

export const List = styled.ul`
  display: grid;
  grid-template-columns: repeat(3, 1fr);
  grid-gap: 20px;
  list-style: none;

  @media (max-width: 900px) {
    grid-template-columns: repeat(2, 1fr);
    grid-gap: 10px;
  }

  li {
    display: flex;
    flex-direction: column;
    background: #fff;
    border-radius: 4px;
    padding: 20px;

    @media (max-width: 900px) {
      padding: 10px;
    }

    img {
      align-self: center;
      max-width: 200px;
      @media (max-width: 900px) {
        max-width: 100px;
      }
    }

    > strong {
      font-size: 16px;
      line-height: 20px;
      margin-top: 5px;
      color: #333;
    }

    > span {
      font-size: 21px;
      font-weight: bold;
      margin: 5px 0 20px;
      @media (max-width: 900px) {
        font-size: 15px;
        margin: 5px 0 10px;
      }
    }

    button {
      background: ${(props) => props.theme};
      border: 0;
      border-radius: 4px;
      overflow: hidden;
      margin-top: auto;

      display: flex;
      align-items: center;
      padding: 12px;
      transition: background 0.2s;

      @media (max-width: 900px) {
        padding: 5px;
      }

      &:hover {
        background: ${(props) => darken(0.03, props.theme)};
      }

      span {
        flex: 1;
        text-align: center;
        font-weight: bold;
        color: #fff;

        @media (max-width: 900px) {
          font-size: 12px;
        }
      }
    }
  }
`;
