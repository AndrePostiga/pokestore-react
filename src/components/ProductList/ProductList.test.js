import React from 'react';
import renderer from 'react-test-renderer'
import 'jest-styled-components'
import { shallow, mount, render } from 'enzyme';


import { ProductList } from './index';
import { List } from './style'

describe('ProductList Component', () => {

  let props;
  beforeEach(() => {
    props = {
      theme: {
        theme: 'grass',
        color: 'green'
      },
      pokemons: [
        {
          id: 1,
          name: 'Test Name',
          urlImage: 'https://test.test',
          price: 23.9,
          formatedPrice: 'R$ 23,90',
          amount: 1
        },
        {
          id: 2,
          name: 'Test Name',
          urlImage: 'https://test.test',
          price: 23.9,
          formatedPrice: 'R$ 23,90',
          amount: 1
        },
      ]
    }
  });

  it('Should render without errors', () => {
    const component = shallow(<ProductList {...props}/>)
    expect(component).toMatchSnapshot()
  })

  it('Should render two items', () => {
    const component = mount(<ProductList {...props}/>)
    expect(component.find('li')).toHaveLength(2)
  })

  // it('Should render item with right color', () => {
  //   const component = mount(<ProductList {...props}/>)
  //   expect(component.find('button').at(1)).toHaveStyleRule('background', props.theme.color)
  // })

});
