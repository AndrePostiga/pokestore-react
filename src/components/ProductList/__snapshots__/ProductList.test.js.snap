// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductList Component Should render without errors 1`] = `ShallowWrapper {}`;
